Fix ackermann steering odometry

franzrammerstorfer opened the pull request:
This is IMHO a quick and easy fix for 

https://github.com/ros-controls/ros2_controllers/issues/878

christophfroehlich reviewed (changes requested):
Hi @franzrammerstorfer! Thanks for the PR. 

Some first comments from my side:

To fix the linter issues, you can check it locally by
- install pre-commit by `pip3 install pre-commit`
- `pre-commit run--all`

Why do you use `EXPECT_FLOAT_EQ` instead of `EXPECT_DOUBLE_EQ`?

Does it make sense to check these cases also with `update_from_velocity` and `update_from_position`, i.e., for `params_.open_loop=false`?

franzrammerstorfer commented:
Hi @christophfroehlich!
Sorry for not listen to pre-commit checks, I did not read the instructions carefully enough. I think I fixed that with my last few commits. I misused EXPECT_FLOAT_EQ, bad habit, thanks for the hint.

Yes, you're right, there are a lot of things open to test -- but right now I just wanted to provide a quick fix for the Ackermann odometry, only rudimentary tested. Maybe we should open a new issue, tagged "help wanted" for proper testing of the steering lib -- starting with tests for the integrator functions, etc. But maybe first we should check for a redesign cf. #692

BTW, I just added this file 'test_steering_odometry.cpp', with some hard-coded tests. Is that the way to go, or do we want to have a more advanced testing style?

christophfroehlich commented:
> Yes, you're right, there are a lot of things open to test -- but right now I just wanted to provide a quick fix for the Ackermann odometry, only rudimentary tested. Maybe we should open a new issue, tagged "help wanted" for proper testing of the steering lib -- starting with tests for the integrator functions, etc. But maybe first we should check for a redesign cf. #692

ok, I'll create a new issue then.

> BTW, I just added this file 'test_steering_odometry.cpp', with some hard-coded tests. Is that the way to go, or do we want to have a more advanced testing style?

Should the expectations of the FW kinmatics be true for all the different kinematic configurations? Then we could change this to a parameterized test as we have with JTC
https://github.com/ros-controls/ros2_controllers/blob/1d0d7531ef8d24ea52e761a872b8bf277652dd33/joint_trajectory_controller/test/test_trajectory_controller_utils.hpp#L695-L710

https://github.com/ros-controls/ros2_controllers/blob/1d0d7531ef8d24ea52e761a872b8bf277652dd33/joint_trajectory_controller/test/test_trajectory_controller.cpp#L1840-L1849

christophfroehlich commented:
But we can for now just fix the CI and extend the tests in a follow-up PR.

christophfroehlich commented:
@aleksandrsizmailovs and @roncapat, as you have commented on the issue: Is this PR fine for you?

christophfroehlich reviewed (approved):
LGTM

roncapat commented:
Seems fine IMHO :)

aleksandrsizmailovs commented:
> @aleksandrsizmailovs and @roncapat, as you have commented on the issue: Is this PR fine for you?

Looks good to me.

bmagyar commented on steering_controllers_library/test/test_steering_odometry.cpp:
```diff
@@ -0,0 +1,114 @@
+  auto cmd1 = std::get<1>(cmd);  // steer
+  EXPECT_LT(cmd1[0], cmd1[1]);   // right (outer) < left (inner)
+  EXPECT_GT(cmd1[0], 0);
+  // std::cout << "\nsteer right (outer)" << cmd1[0] << std::endl;
+  // std::cout << "\nsteer left (inner)" << cmd1[1] << std::endl;
```
```suggestion
```

bmagyar commented on steering_controllers_library/test/test_steering_odometry.cpp:
```diff
@@ -0,0 +1,114 @@
+  auto cmd1 = std::get<1>(cmd);                     // steer
+  EXPECT_GT(std::abs(cmd1[0]), std::abs(cmd1[1]));  // abs right (inner) > abs left (outer)
+  EXPECT_LT(cmd1[0], 0);
+  // std::cout << "\nsteer right (inner)" << cmd1[0] << std::endl;
+  // std::cout << "\nsteer left (outer)" << cmd1[1] << std::endl;
```
```suggestion
```
